modules/abstract-eth/src/lib/transferBuilder.ts

```diff
@@ -137,11 +143,12 @@ export class TransferBuilder {
 
   protected getOperationData(): (string | Buffer)[][] {
     let operationData;
+
     if (this._tokenContractAddress !== undefined) {
       operationData = [
         ['string', 'address', 'uint', 'address', 'uint', 'uint'],
         [
-          this.getTokenOperationHashPrefix(),
+          this._tokenOperationHashPrefix ?? this.getTokenOperationHashPrefix(),
```

Comment:
Could this check be in `getTokenOperationHashPrefix()` instead?

Reply:
Basically we need coin and network to derive this value because now as we are using chain id. We are overriding getTokenOperationHashPrefix() method in the coin specific modules like sdk-coin-eth, sdk-coin-polygon etc. So we will have to add this similar logic in all the individual coin's modules. Hence I thought to do it at this one place only. The reason why I added this method call as well in this condition, is to make it backwards compatible

Reply:
Actually, this might not be applicable for those coins because they won't have chain Id changes right so need to keep the override as is anyways? For coins which we have chainId changes, we don't need to overrirde them as the changes here should be sufficient as we are not using coin specific names

Reply:
Yeah, override is not needed for the new coins, it is just for needed for older coins

Reply:
Have updated this.